src/finalise_field.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ Crown Copyright 2012 AWE.
+
+ This file is part of CloverLeaf.
+
+ CloverLeaf is free software: you can redistribute it and/or modify it under 
+ the terms of the GNU General Public License as published by the 
+ Free Software Foundation, either version 3 of the License, or (at your option) 
+ any later version.
+
+ CloverLeaf is distributed in the hope that it will be useful, but 
+ WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or 
+ FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more 
+ details.
+
+ You should have received a copy of the GNU General Public License along with
+ CloverLeaf. If not, see http://www.gnu.org/licenses/.
+ */
+
+
+// @brief  Allocates the data for each mesh chunk
+// @author Wayne Gaudin
+// @details The data fields for the mesh chunk are allocated based on the mesh
+// size.
+
+
+#include "finalise_field.h"
+
+
+// Allocate Kokkos Views for the data arrays
+void finalise_field(global_variables &globals) {
+
+	// XXX GCC 10 crashes with target exit data call, so the program terminates with a
+	// non zero exit code, event if correct results are produced
+	// only destroy context for non GCC compilers now
+	#if !defined(__GNUC__) || !defined(__GNUG__)
+
+	for (int tile = 0; tile < globals.config.tiles_per_chunk; ++tile) {
+
+		tile_type &t = globals.chunk.tiles[tile];
+		field_type &field = t.field;
+
+		#pragma omp target exit data \
+				map(release: field.density0.data[:0]) \
```

Comment:
Does GCC crash if you just pass in the name `field.density0.data` rather than the array section `[:0]`?

Reply:
Resolved in https://github.com/UoB-HPC/cloverleaf_openmp_target/pull/2/commits/c70d7eb61ed7738ef597cf35043e89748aeda70e